ambry-router/src/main/java/com.github.ambry.router/DeleteManager.java

```diff
@@ -134,10 +135,12 @@ void handleResponse(ResponseInfo responseInfo) {
     DeleteOperation deleteOperation = correlationIdToDeleteOperation.remove(correlationId);
     // If it is still an active operation, hand over the response. Otherwise, ignore.
     if (deleteOperations.contains(deleteOperation)) {
-      deleteOperation.handleResponse(responseInfo);
+      deleteOperation.handleResponse(responseInfo, false);
       if (deleteOperation.isOperationComplete() && deleteOperations.remove(deleteOperation)) {
         onComplete(deleteOperation);
       }
+    } else {
+      deleteOperation.handleResponse(responseInfo, true);
```

Comment:
Do we have to not ignore a response just because of metrics? Why care about the metrics of a response we do not care about? Would rather just have a metric for ignored responses - that is the number of responses that come in after  operation completes.